archive/tar.go

```diff
@@ -107,6 +107,12 @@ func Apply(ctx context.Context, root string, r io.Reader) (int64, error) {
 
 	// Iterate through the files in the archive.
 	for {
+		select {
+		case <-ctx.Done():
+			return 0, ctx.Err()
```

Comment:
is there any cleanup that needs to be done?

Reply:
This is relying on `ctx.Err()` returning a non-nil error and that triggering cleanup by the caller. The caller is responsible for closing the reader and unmounting. The client is responsible for deleting any snapshots that were created for apply.